### ISSUE TITLE
Silent inconsistency for dayofyear for non-standard calendar

### ISSUE THREAD

**Issue by Balinus** [Alexander-Barth/NCDatasets.jl#29]
Hello!

Stumbled into some inconsistency with the non-standard calendar in terms of `dayofyear`.

MWE
```julia
using Dates, NCDatasets
julia> daysinyear(DateTimeNoLeap(2008))
365

julia> dayofyear(DateTimeNoLeap(2008, 12, 31))
366
```

As far as I understand, the method `dayofyear` is implemented in `Dates` only and reexported in NCDatasets.

Cheers!

**Comment by Alexander-Barth** [comment]
Thank you for reporting this issue.
`dayofyear` is indeed just implemented in `Dates`, but `DateTimeNoLeap` derives from `Dates.TimeType` therefore, we have this silent error (while previously it would have generated an error message).

```julia
@which dayofyear(DateTimeNoLeap(2008, 12, 31))
dayofyear(dt::TimeType) in Dates at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Dates/src/query.jl:309
```

I had to derive the date types like `DateTimeNoLeap` from  `Dates.TimeType`  in order to implement more flexible parsing methods. But unfortunately, this results in this type of silient errors.

To solve this issue, I have now implemented `dayofyear`, `firstdayofyear` in NCDatasets for all `AbstractCFDateTime`s.

Does this solve the issue for you?
